assets/src/edit-story/app/story/storyProvider.js

```diff
@@ -101,9 +104,10 @@ function StoryProvider({ storyId, children }) {
     );
 
     return {
-      selectedElementIds: selection,
-      selectedElements: els,
-      selectedElementAnimations: animations,
+      selectedElementIds: selection.length > 0 ? selection : EMPTY_ARRAY,
+      selectedElements: els.length > 0 ? els : EMPTY_ARRAY,
+      selectedElementAnimations:
+        animations.length > 0 ? animations : EMPTY_ARRAY,
```

Comment:
This should only fix the shallow compare where selection is empty, not when an element is being dragged around right?

Reply:
In my testing `selectedElementAnimations` was always a _different_ empty array when moving an element around, causing the equality checks to fail. But the element always remained selected, so not related to selection.

Reply:
Ah I see. 

Alternatively we could memoize the generation of `selectedElementAnimations`, though the enclosing function is memoized on `[currentPage, selection]`, so is the larger issue that `currentPage` or `selection` is changing on drag?

Reply:
It's `currentPage` that is changing, because it contains `elements`, which always changes due to the element changing position.